tools/docker/quorum-multi-party-all-in-one/Dockerfile

```diff
@@ -1,116 +1,54 @@
-FROM docker:20.10.3-dind
-
-ARG BESU_VERSION=21.1.2
-ARG QUORUM_VERSION=21.4.1
-ARG QUORUM_TESSERA_VERSION=21.1.1
-ARG CA_VERSION=1.4.9
-
-WORKDIR /
-
-RUN apk update
-
-# Install dependencies of Docker Compose
-RUN apk add py-pip python3-dev libffi-dev openssl-dev gcc libc-dev make
-
-# Install python/pip - We need this because DinD 18.x has Python 2
-# And we cannot upgrade to DinD 19 because of
-# https://github.com/docker-library/docker/issues/170
-ENV PYTHONUNBUFFERED=1
-RUN apk add --update --no-cache python3 && ln -sf python3 /usr/bin/python
-RUN python3 -m ensurepip
-RUN pip3 install --no-cache --upgrade "pip>=21" setuptools
-
-# Without this the docker-compose installation crashes, complaining about
-# a lack of rust compiler...
-# RUN pip install setuptools_rust
-ENV CRYPTOGRAPHY_DONT_BUILD_RUST=1
-
-# Install Docker Compose which is a dependency of Fabric Samples
-RUN pip install docker-compose
-
-# Need git to clone the sources of the Fabric Samples repository from GitHub
-RUN apk add --no-cache git
-
-# Fabric Samples needs bash, sh is not good enough here
-RUN apk add --no-cache bash
-
-# The file binary is used to inspect exectubles when debugging container image issues
-RUN apk add --no-cache file
-
-# Need NodeJS tooling for the Typescript contracts
-RUN apk add --no-cache npm nodejs
-
-# Needed because the Fabric binaries need the GNU libc dynamic linker to be executed
-# and alpine does not have that by default
-# @see https://askubuntu.com/a/1035037/1008695
-# @see https://github.com/gliderlabs/docker-alpine/issues/219#issuecomment-254741346
-RUN apk add --no-cache libc6-compat
-
-RUN apk add --no-cache --update chromium
-
-ENV CACTUS_CFG_PATH=/etc/hyperledger/cactus
-RUN mkdir -p $CACTUS_CFG_PATH
-# OpenSSH - need to have it so we can shell in and install/instantiate contracts
-RUN apk add --no-cache openssh augeas
-
-# Configure the OpenSSH server we just installed
-RUN augtool 'set /files/etc/ssh/sshd_config/AuthorizedKeysFile ".ssh/authorized_keys /etc/authorized_keys/%u"'
-RUN augtool 'set /files/etc/ssh/sshd_config/PermitRootLogin yes'
-RUN augtool 'set /files/etc/ssh/sshd_config/PasswordAuthentication no'
-RUN augtool 'set /files/etc/ssh/sshd_config/PermitEmptyPasswords no'
-RUN augtool 'set /files/etc/ssh/sshd_config/Port 22'
-RUN augtool 'set /files/etc/ssh/sshd_config/LogLevel DEBUG2'
-RUN augtool 'set /files/etc/ssh/sshd_config/LoginGraceTime 10'
-# Create the server's key - without this sshd will refuse to start
-RUN ssh-keygen -A
+################################
```

Comment:
@outSH The new image supresses the logs from the inner containres that print out this very useful list of hosts & ports for debugging & development purposes, could you please make sure that the output is still there in the new image. If you run the old one, it prints these at the end when the booting of the network has succeeded, I'm including an example of it here:

```sh
*************************************
Quorum Dev Quickstart 
*************************************
2022-03-30 19:42:08,192 DEBG 'quorum-network' stdout output:
*************************************
Quorum Dev Quickstart 
*************************************

----------------------------------
List endpoints and services
----------------------------------
2022-03-30 19:42:10,623 DEBG 'quorum-network' stdout output:
----------------------------------
List endpoints and services
----------------------------------

JSON-RPC HTTP service endpoint                 : http://localhost:8545
JSON-RPC WebSocket service endpoint            : ws://localhost:8546
Web block explorer address                     : http://localhost:25000/
2022-03-30 19:42:10,624 DEBG 'quorum-network' stdout output:
JSON-RPC HTTP service endpoint                 : http://localhost:8545
JSON-RPC WebSocket service endpoint            : ws://localhost:8546
Web block explorer address                     : http://localhost:25000/

Prometheus address                             : http://localhost:9090/graph
2022-03-30 19:42:11,910 DEBG 'quorum-network' stdout output:
Prometheus address                             : http://localhost:9090/graph

Cakeshop toolkit address                       : http://localhost:8999
2022-03-30 19:42:13,285 DEBG 'quorum-network' stdout output:
Cakeshop toolkit address                       : http://localhost:8999

Grafana address                                : http://localhost:3000/d/a1lVy7ycin9Yv/goquorum-overview?orgId=1&refresh=10s&from=now-30m&to=now&var-system=All

For more information on the endpoints and services, refer to README.md in the installation directory.
****************************************************************
2022-03-30 19:42:14,553 DEBG 'quorum-network' stdout output:
Grafana address                                : http://localhost:3000/d/a1lVy7ycin9Yv/goquorum-overview?orgId=1&refresh=10s&from=now-30m&to=now&var-system=All

For more information on the endpoints and services, refer to README.md in the installation directory.
****************************************************************

```

Reply:
@petermetz These messages come directly from `Quorum Dev Quickstart` which is started the same way as before, I've double checked and the logs are still there.

Reply:
@outSH Weird stuff, let's try to figure it out together then because I triple checked it and it's not there. Here's the command I ran and the log file. I did a full text search in the log file and the endpoint information is not in there. 

What command did you run to build the image? What command to run it? Can you also attach the logs so we can compare?
Maybe I didn't build the image right/same way you did or something.

`docker run --rm --privileged ghcr.io/hyperledger/cactus-quorum-multi-party-all-in-one:2022-03-30--1928 &> container.log`

[container.log](https://github.com/hyperledger/cactus/files/8399720/container.log)

Reply:
@petermetz Ooh, I see, this is weird indeed...

Basically the logs appear again when you add `-ti` flag, without this it complains about `No value for $TERM and no -T specified`. I think it might be caused by the fact that I use newer sample app (from npm, not git). One difference I see is that the new script writes it's messages to stderr instead of stdout. Nevertheless, it should work the same, both streams should be piped through supervisord and it didn't cause any problems before, so I have no idea what's happening now, or how to solve it without digging into quorum code :/

Reply:
@outSH A workaround is fine (instead of fixing it upstream in the quorum repo itself) but we need to make sure that people see the endpoint list when using the image as documented/instructed. 

From what you are saying my guess is that you could get away with 
1. Updating readme of the image so that every mention of the `docker run ...` command in there contains the `--tty` flag (leave out the --interactive flag in my opinion it's all about having `--tty` on, the `-i` flag makes no difference in this case so less confusing if not included in the examples)
2. Add a little note in there explaining that the `--tty` is required by the underlying software that we are running inside the container, without the `--tty` flag the endpoint list will not be printed in the logs of the container. --- This should preemptively take care of people getting confused or struggling to work with the image.

Reply:
@petermetz Done, I've updated the README